packages/tx/src/4844/tx.ts

```diff
@@ -187,10 +187,26 @@ export class Blob4844Tx implements TransactionInterface<typeof TransactionType.B
       }
     }
 
+    // EIP-7594 PeerDAS: Limit of 6 blobs per transaction
+    if (this.common.isActivatedEIP(7594)) {
+      const maxBlobsPerTx = this.common.param('maxBlobsPerTx')
+      if (this.blobVersionedHashes.length > maxBlobsPerTx) {
+        const msg = Legacy.errorMsg(
+          this,
+          `tx can contain at most ${maxBlobsPerTx} blobs (EIP-7594)`,
+        )
+        throw EthereumJSErrorWithoutCode(msg)
+      }
+    }
+
+    // "Old" limit (superseded by EIP-7594 starting with Osaka)
     const limitBlobsPerTx =
       this.common.param('maxBlobGasPerBlock') / this.common.param('blobGasPerBlob')
     if (this.blobVersionedHashes.length > limitBlobsPerTx) {
-      const msg = Legacy.errorMsg(this, `tx can contain at most ${limitBlobsPerTx} blobs`)
+      const msg = Legacy.errorMsg(
+        this,
+        `tx can contain at most ${limitBlobsPerTx} blobs (maxBlobGasPerBlock/blobGasPerBlob)`,
```

Comment:
This error message does not look completely correct, because `maxBlobGasPerBlock/blobGasPerBlob` could be 9 (osaka config: target 6 but max 9), while tx limit is 6